From 6e9213054338791936d7211c1049d731f9980ba8 Mon Sep 17 00:00:00 2001
From: Scokart Gilles <gscokart@apache.org>
Date: Fri, 7 Dec 2007 08:40:56 +0000
Subject: [PATCH] FIX: NPE in namespace transformation during the
 ivy:findrevision and ivy:resolve task execution (IVY-659) (thanks to Andrea
 Bernardo Ciddio)

git-svn-id: https://svn.apache.org/repos/asf/ant/ivy/core/trunk@602022 13f79535-47bb-0310-9956-ffa450edef68
---
 CHANGES.txt                                              | 2 ++
 src/java/org/apache/ivy/plugins/namespace/Namespace.java | 6 ++++++
 2 files changed, 8 insertions(+)

diff --git a/CHANGES.txt b/CHANGES.txt
index 7f2364667..a383ef8c5 100644
--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -19,6 +19,7 @@ for detailed view of each issue, please consult http://issues.apache.org/jira/br
 	Jim Bonanno
 	Matthieu Brouillard
 	Kristian Cibulskis
+	Andrea Bernardo Ciddio
 	Jacob Grydholt Jensen
 	Scott Goldstein
 	Pierre Hägnestrand
@@ -57,6 +58,7 @@ for detailed view of each issue, please consult http://issues.apache.org/jira/br
    version in SVN
 =====================================
 - FIX: Strict conflictmanager seems to not support dynamic revisions (IVY-474)
+- FIX: NPE in namespace transformation during the ivy:findrevision and ivy:resolve task execution (IVY-659) (thanks to Andrea Bernardo Ciddio)
 
 
    2.0.0-beta1
diff --git a/src/java/org/apache/ivy/plugins/namespace/Namespace.java b/src/java/org/apache/ivy/plugins/namespace/Namespace.java
index ba3f23681..e32655dea 100644
--- a/src/java/org/apache/ivy/plugins/namespace/Namespace.java
+++ b/src/java/org/apache/ivy/plugins/namespace/Namespace.java
@@ -37,6 +37,9 @@ public class Namespace {
 
     private NamespaceTransformer fromSystemTransformer = new NamespaceTransformer() {
         public ModuleRevisionId transform(ModuleRevisionId mrid) {
+            if (mrid == null) {
+                return null;
+            }
             for (Iterator iter = rules.iterator(); iter.hasNext();) {
                 NamespaceRule rule = (NamespaceRule) iter.next();
                 ModuleRevisionId nmrid = rule.getFromSystem().transform(mrid);
@@ -56,6 +59,9 @@ public boolean isIdentity() {
 
     private NamespaceTransformer toSystemTransformer = new NamespaceTransformer() {
         public ModuleRevisionId transform(ModuleRevisionId mrid) {
+            if (mrid == null) {
+                return null;
+            }
             for (Iterator iter = rules.iterator(); iter.hasNext();) {
                 NamespaceRule rule = (NamespaceRule) iter.next();
                 ModuleRevisionId nmrid = rule.getToSystem().transform(mrid);
